torchbenchmark/models/gen_timm_models.py

```diff
@@ -0,0 +1,149 @@
+import os
+import inspect
+from pathlib import Path
+
+timm_model_cfgs = [
+    ('nfnet', 'CLASSIFICATION', 'dm_nfnet_f0'),
+    ('efficientnet', 'CLASSIFICATION', 'mixnet_m'),
+    ('regnet', 'CLASSIFICATION', 'regnety_002'),
+    ('resnest', 'CLASSIFICATION', 'resnest14d'),
+    ('vision_transformer', 'GENERATION', 'vit_small_patch16_224'),
+    ('vovnet', 'DETECTION', 'vovnet39a'),
+]
+
+init_program_template = """# Generated by gen_timm_models.py
+import torch
+import timm.models.{model_name}
+
+from ...util.model import BenchmarkModel
+from torchbenchmark.tasks import COMPUTER_VISION
+from .config import TimmConfig
+
+class Model(BenchmarkModel):
+    task = COMPUTER_VISION.{model_category}
+
+    def __init__(self, device=None, jit=False, variant='{model_variant}', precision='float32'):
+        super().__init__()
+        self.device = device
+        self.jit = jit
+        self.model = timm.create_model(variant, pretrained=False, scriptable=True)
+        self.cfg = TimmConfig(model = self.model, device = device, precision = precision)
+        self.model.to(
+            device=self.device,
+            dtype=self.cfg.model_dtype
+        )
+        if device == 'cuda':
+            torch.cuda.empty_cache()
+        if jit:
+            self.model = torch.jit.script(self.model)
```

Comment:
Do you know how many of these models actually script?

Reply:
Sorry I don't. How to verify that the model actually scripts?

Reply:
The [benchmarking code](https://github.com/rwightman/pytorch-image-models/blob/master/benchmark.py#L166) from timms package shows all models created in the repository should be able to jit.

Reply:
if the line of code does not crash, then it scripts. but you might want to make sure it's actually hitting that line in one case if you're not sure.  (you can also confirm that self.model is a 'scriptmodule' object rather than a 'module' object.

Reply:
Makes sense, added the jit script check.

Reply:
oh- to be clear, the typecheck isn't necessary.  the only reason i suggested that is if you wanted to convince yourself in the debugger.  as long as you know the line of code is reached, it should be fine without the additional check.  (doesn't hurt anything to have it either though)